python/cudf/cudf/tests/test_datetime.py

```diff
@@ -1306,8 +1306,11 @@ def test_datetime_infer_format(data, timezone, dtype):
 
         assert_eq(expected, actual)
     else:
-        with pytest.raises(NotImplementedError):
-            sr.astype(dtype)
+        with cudf.option_context("mode.pandas_compatible", True):
+            with pytest.raises(NotImplementedError):
+                sr.astype(dtype)
+
+        # pandas doesn't allow parsing "Z" to naive type
```

Comment:
Please move this comment above the `sr.astype(dtype)` call to make it clear that it's the cause of the error.

Reply:
Sure thing. Moved